cla_backend/apps/knowledgebase/tests/test_csv_2_fixture.py

```diff
@@ -28,1416 +47,724 @@ def slice_records(self, output_records, expected_records, start_index=None, end_
+                        u"article": 1,
+                        u"article_category": 18,
+                        u"preferred_signpost": True,
+                    },
+                },
+            ]
+            outputList = json.loads(outputJSON)
+            output_acm = filter(lambda x: x["model"] == "knowledgebase.articlecategorymatrix", outputList)
+            self.compare_length(output_acm, expectedList)
+
+            pk_range_for_acm = self.calculate_pk_range(expectedList)
+
+            output_sorted_acm, expected_sorted_acm = self.sort_article_category_matrices(output_acm, expectedList)
+            self.compare_article_category_matrices(output_sorted_acm, expected_sorted_acm, pk_range_for_acm)
 
-        self.compare_length(outputList, expectedList)
+    def test_fixture_with_multiple_articles(self):
+        csv_file = self.find_path_to_csvfile("multiple_articles")
+        file = open(csv_file)
+        try:
+            csv = KnowledgebaseCsvParse(file)
+        except SystemExit:
+            self.fail("Missing required fields")
+        else:
+            outputJSON = csv.fixture_as_json()
+            outputList = json.loads(outputJSON)
 
-        output_article_categories, expected_article_categories = self.slice_records(
-            outputList, expectedList, end_index=18
-        )
-        self.compare_list_of_records(output_article_categories, expected_article_categories)
+            sorted_records = defaultdict(list)
 
-        output_article_1, expected_article_1 = self.slice_records(outputList, expectedList, 18, 19)
-        self.compare_list_of_records(output_article_1, expected_article_1)
+            for record in outputList:
+                sorted_records[record["model"]].append(record)
 
-        output_acm_1, expected_acm_1 = self.slice_records(outputList, expectedList, 19, 37)
-        pk_range_acm_1 = self.calculate_pk_range(output_acm_1)
-        output_sorted_acm_1, expected_sorted_acm_1 = self.sort_article_category_matrices(output_acm_1, expected_acm_1)
-        self.compare_article_category_matrices(output_sorted_acm_1, expected_sorted_acm_1, pk_range_acm_1)
+            article_category_records = sorted_records["knowledgebase.articlecategory"]
+            self.assertEqual(len(article_category_records), 18)
 
-        output_article_2, expected_article_2 = self.slice_records(outputList, expectedList, 37, 38)
-        self.compare_list_of_records(output_article_2, expected_article_2)
+            article_records = sorted_records["knowledgebase.article"]
+            article_1, article_2 = article_records
 
-        output_article_2_category_matrix = outputList[-18:]
-        expected_article_2_category_matrix = expectedList[-18:]
+            self.assertEqual(len(article_records), 2)
+            self.assertNotEqual(article_1["fields"]["website"], article_2["fields"]["website"])
+            self.assertNotEqual(article_1["pk"], article_2["pk"])
```

Comment:
Whereas I think this is a good assertion, because checking they're different from each other makes sense with no further information